File: app/src/main/java/com/test/ashish/githubissues/IssuesAdapter.java
package com.test.ashish.githubissues;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.test.ashish.githubissues.Pojo.Issues;

import java.util.List;

public class IssuesAdapter extends RecyclerView.Adapter<IssuesAdapter.ViewHolder> {

    private List<Issues> mItems;
    private Context mContext;

    public class ViewHolder extends RecyclerView.ViewHolder {

        private TextView prNumberTextView;
        private TextView titleTextView;
        private TextView userTextView;
        private TextView patchUrlTextView;
        private TextView stateTextView;

        public ViewHolder(View itemView) {
            super(itemView);

            prNumberTextView = (TextView) itemView.findViewById(R.id.prNumber);
            titleTextView = (TextView) itemView.findViewById(R.id.title);
            userTextView = (TextView) itemView.findViewById(R.id.userName);
            patchUrlTextView = (TextView) itemView.findViewById(R.id.patch_url);
            stateTextView = (TextView) itemView.findViewById(R.id.state);

        }

    }

    public IssuesAdapter(Context context, List<Issues> issues) {

        this.mItems = issues;
        this.mContext = context;

    }

    @Override
    public IssuesAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {

        Context context = parent.getContext();
        LayoutInflater inflater = LayoutInflater.from(context);

        View postView = inflater.inflate(R.layout.issues_item, parent, false);

        ViewHolder viewHolder = new ViewHolder(postView);
        return viewHolder;

    }

    @Override
    public void onBindViewHolder(IssuesAdapter.ViewHolder holder, int position) {

        Issues item = mItems.get(position);
        holder.prNumberTextView.setText(String.valueOf(item.getPrNumber()));
        holder.titleTextView.setText(item.getTitle());
        holder.userTextView.setText(item.getUser());
        holder.patchUrlTextView.setText(item.getPatchUrl());
        holder.stateTextView.setText(item.getState());
    }

    @Override
    public int getItemCount() {
        return mItems.size();
    }

    private Issues getItem(int adapterPosition) {
        return mItems.get(adapterPosition);
    }

    public void resetList(List<Issues> issues){
        mItems = issues;
        notifyDataSetChanged();
    }


}

File: app/src/main/java/com/test/ashish/githubissues/IssuesAdapter.kt
package com.test.ashish.githubissues

import android.content.Context
import com.test.ashish.githubissues.Pojo.Issues
import android.widget.TextView
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import androidx.recyclerview.widget.RecyclerView

class IssuesAdapter(private val mContext: Context, private var mItems: List<Issues?>) :
    RecyclerView.Adapter<IssuesAdapter.ViewHolder>() {
    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val prNumberTextView: TextView
        val titleTextView: TextView
        val userTextView: TextView
        val patchUrlTextView: TextView
        val stateTextView: TextView

        init {
            prNumberTextView = itemView.findViewById<View>(R.id.prNumber) as TextView
            titleTextView = itemView.findViewById<View>(R.id.title) as TextView
            userTextView = itemView.findViewById<View>(R.id.userName) as TextView
            patchUrlTextView = itemView.findViewById<View>(R.id.patch_url) as TextView
            stateTextView = itemView.findViewById<View>(R.id.state) as TextView
        }
    }

    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): ViewHolder {
        val context = parent.context
        val inflater = LayoutInflater.from(context)
        val postView = inflater.inflate(R.layout.issues_item, parent, false)
        return ViewHolder(postView)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val item = mItems[position]
        holder.prNumberTextView.text = item!!.prNumber.toString()
        holder.titleTextView.text = item!!.title
        holder.userTextView.text = item!!.user
        holder.patchUrlTextView.text = item!!.patchUrl
        holder.stateTextView.text = item!!.state
    }

    override fun getItemCount(): Int {
        return mItems.size
    }

    private fun getItem(adapterPosition: Int): Issues? {
        return mItems[adapterPosition]
    }

    fun resetList(issues: List<Issues?>) {
        mItems = issues
        notifyDataSetChanged()
    }
}